src/synopsys-action/tools-parameter.ts

```diff
@@ -304,7 +305,7 @@ export class SynopsysToolsParameter {
     }
 
     // This condition is required as per ts-lint as these fields may have undefined as well
-    if (githubRepoName != null && githubBranchName != null && githubRepoOwner != null) {
+    if (githubHostUrl) {
```

Comment:
As @spurohitsynopsys mentioned, if githubHostUrl is empty, other github data will not set. And other parameters are coming from github api which will not be undefined so i just default it to an empty string instead of undefined.

Reply:
but this is not mandatory, it can be empty that you have to set it in above condition if GITHUB_SERVER_URL returns https://github.com

Reply:
`const githubHostUrl = githubServerUrl === constants.GITHUB_CLOUD_URL ? undefined : githubServerUrl`
Here it will be undefined. So i think it meets the condition.